src/task/period.rs

```diff
@@ -0,0 +1,91 @@
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde-support", derive(Serialize, Deserialize))]
+pub enum Period {
+    Day,
+    Week,
+    Month,
+    Year,
+}
+
+impl Period {
+    fn is_leap_year(year: i32) -> bool {
+        year % 4 == 0 && (year % 100 != 0 || year % 400 == 0)
+    }
+}
+
+impl ::std::str::FromStr for Period {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, ()> {
+        use self::Period::*;
+
+        match s {
+            "d" => Ok(Day),
+            "w" => Ok(Week),
+            "m" => Ok(Month),
+            "y" => Ok(Year),
+            _ => Err(()),
+        }
+    }
+}
+
+impl ::std::fmt::Display for Period {
+    fn fmt(&self, f: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
+        use self::Period::*;
+
+        let s = match *self {
+            Day => "d",
+            Week => "w",
+            Month => "m",
+            Year => "y",
+        };
+
+        f.write_str(s)?;
+
+        Ok(())
+    }
+}
+
+impl ::std::ops::Add<::chrono::NaiveDate> for Period {
+    type Output = ::chrono::NaiveDate;
+
+    fn add(self, rhs: Self::Output) -> Self::Output {
+        use self::Period::*;
```

Comment:
I think it would be safer and simpler to handle adding days using `rhs + chrono::Duration::days(d)` instead of doing it by hand.

Reply:
Cf. my other comment, I think you could do it like this: https://github.com/Ekleog/todiff/blob/c11b62d0cc26d1a67211ca789ec16292aafd045e/src/compute_changes.rs#L124-L137

Reply:
Hmm, however it seems we can't handle something like `x todo due:2010-01-31 rec:+1m`. Your approach may be the only one that works...